Lessons_18/task1.py

```diff
@@ -0,0 +1,102 @@
+"""Task 1
+Create a class method named 'validate', which should be called from the 
+'__init__' method to validate parameter email, passed to the constructor.
+The logic inside the 'validate' method could be to check if the passed email
+parameter is a valid email string.
+"""
+
+
+class User:
+    """Create user object"""
+
+    def __init__(self, login, name, email):
+        self.login = login
+        self.name = name
+        self.email = User.validate(email) or "Wrong email format!"
+
+    def talk(self):
+        """Print user greeting"""
+        print(
+            f"Hello, I am {self.name}! My user login {self.login}.\n"
+            f"You can send me email: {self.email}"
+        )
+
+    @staticmethod
+    def email_to_part(email):
+        """Return tuple that includes prefix, and two part of domain from email address"""
+        if email.count("@"):
+            sep_index = email.index("@")
+            prefix = email[0:sep_index]
+            domain = email[sep_index + 1:]
+            if domain.count("."):
+                point_index = domain.index(".")
+                domain_name = domain[:point_index]
+                top_level_domain = domain[point_index + 1:]
+                if (
+                    len(prefix) > 2
+                    and len(domain_name) > 0
+                    and len(top_level_domain) >= 2
+                ):
+                    return prefix, domain_name, top_level_domain
+        return None, None, None
+
+    @staticmethod
+    def check_email_part(part, symbols_list, with_num=True):
+        """Check if the passed email part is a valid string. Return True or False"""
+        def valid_char(char):
```

Comment:
There may be also different symbols in the first part of the email, like `+` and `-`